main/webapp/modules/core/scripts/views/data-table/cell-ui.js

```diff
@@ -604,6 +604,21 @@ DataTableCellUI.prototype._startEdit = function(elmt) {
   cellDataType = (this._cell !== null && "t" in this._cell && this._cell.t !=  null) ? this._cell.t : cellDataType;
   elmts.typeSelect.val(cellDataType);
 
+  elmts.typeSelect.on('change', function() {
+    var newType = elmts.typeSelect.val();
+    if (newType === "date") {
+      elmts.cell_help_text.html($.i18n('core-views/cell-edit-date-help'));
+      $(elmts.cell_help_text).css({'font-style': 'italic', 'display': ''});
+    } else {
+      $(elmts.cell_help_text).css({'display': 'none'});
+    }
+  });
+
+  if (cellDataType === "date") {
+    elmts.cell_help_text.html($.i18n('core-views/cell-edit-date-help'));
+    $(elmts.cell_help_text).css({'font-style': 'italic', 'display': ''});
```

Comment:
I think we can improve this code a little:
* use `.show()` or `.hide()` rather than explicit calls to `.css(...)` to show and hide an element
* setting up the element (by adding the text and setting its font-style) should be done only once, when setting up the dialog (even if the element is not shown initially). That makes the rest of the code cleaner: when an event happens, you should only need to `.show()` or `.hide()` the element.